### PR TITLE
[PD] Add a Suppressed property to disable a feature

### LINKED ISSUE #12073: [Feature Request] PD: add a Passthrough boolean property to allow to disable a feature
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

We currently can't disable a feature without remove it.
I propose to add a Passthrough boolean property: when true the feature Base shape is copied as the result of the operation, when false the feature compute its own result.

### Full version info

```shell
OS: Debian GNU/Linux trixie/sid (XFCE/xfce)
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.35649 (Git)
Build type: Release
Branch: main
Hash: 35e09e233c4fadb6194c82164ece25d5043d23fe
Python 3.11.7, Qt 5.15.10, Coin 4.0.2, Vtk 9.1.0, OCC 7.6.3
Locale: French/France (fr_FR)
```

### Subproject(s) affected?

PartDesign

### Anything else?

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by FlachyJoe**
Aims to fix #12073

### CODE REVIEW

File: src/Mod/PartDesign/App/FeatureHole.cpp [hunk 1]
```diff
@@ -1664,6 +1664,10 @@ App::DocumentObjectExecReturn* Hole::execute()
     TopoDS_Shape base;
     try {
         base = getBaseShape();
+        if (Passthrough.getValue()) {
+            this->Shape.setValue(base);
+            return App::DocumentObject::StdReturn;
+        }
```
Comment (kadet1090) [inline src/Mod/PartDesign/App/FeatureHole.cpp:1670]: This gets repeated over and over again - is there a better place for this piece of code so it does not have to be duplicated? That way feature developer need to remember to cover this case which is common for every feature and thus should be higher in the hierarchy.

I am not sure however if this is possible at all now as this probably would have to be done on the caller side?

Comment (FlachyJoe) [reply src/Mod/PartDesign/App/FeatureHole.cpp]: `execute()` is called by `DocumentObject::recompute()` It's not a PD specific procedure so we can't tweak this.

Comment (kadet1090) [reply src/Mod/PartDesign/App/FeatureHole.cpp]: Yeah, I checked how "Refine" which is also common property is implemented and well, it is also must be called in all features. So this is fine I suppose, maybe an idea for later refactor. I'm marking this as resolved as it is not a PR that should do a major overhaul of how features are implemented.

**Review by kadet1090 — APPROVED** [review]

File: src/Mod/PartDesign/App/FeatureHole.cpp (hunk 1 above)
Comment (FlachyJoe) [reply src/Mod/PartDesign/App/FeatureHole.cpp]: @kadet1090 I found a solution by override `recompute`

Comment (FlachyJoe) [reply src/Mod/PartDesign/App/FeatureHole.cpp]: It doesn't work as expected, I can edit a existing feature but no more create a new one.
Fall back to working previous but still work on this solution.

**Comment by FEA-eng** [comment]
It would be awesome to have it. Could it work for FEM features too ? In fact, I was going to create a feature request for that.

**Comment by FlachyJoe** [comment]
Rewritten, squashed and rebased. Ready for review.

Side effects:

- if a suppressed feature is used in a pattern, the pattern show its normal result minus the original.
- it's a good manner to reduce a file size.

**Comment by FlachyJoe** [comment]
> Could it work for FEM features too ?

The logic is completely different. Feel free to open a FR.

**Comment by Dutchtallman** [comment]
> > Could it work for FEM features too ?
> 
> The logic is completely different. Feel free to open a FR.

Why is the logic different? ... to me, the high level concept is the same, but lower level implications are different... similar to 'addition' as a high level concept works out differently on lower level: integers or strings... Should there be generic 'suppress feature'?... with different implementations in different workbenches?

**Pull request merged**